go/chat/storage/storage.go

```diff
@@ -1114,6 +1142,53 @@ func (s *Storage) updateUnfurlTargetOnDelete(ctx context.Context, convID chat1.C
 	return chat1.NewMessageUnboxedWithValid(mvalid), nil
 }
 
+func (s *Storage) updateRepliesAffected(ctx context.Context, convID chat1.ConversationID, uid gregor1.UID,
+	replies []chat1.MessageID, replyMap map[chat1.MessageID]chat1.MessageUnboxed) {
+	if len(replies) == 0 {
+		return
+	}
+	defer s.Trace(ctx, func() error { return nil }, "updateRepliesAffected: num: %d", len(replies))()
+	for _, reply := range replies {
+		if _, ok := replyMap[reply]; ok {
+			continue
+		}
+		replyMsg, err := s.getMessage(ctx, convID, uid, reply)
+		if err != nil || replyMsg == nil {
+			s.Debug(ctx, "updateRepliesAffected: failed to get message: err: %s", err)
+			continue
+		}
+		replyMap[reply] = *replyMsg
+	}
+}
+
+func (s *Storage) GetExplodedReplies(ctx context.Context, convID chat1.ConversationID, uid gregor1.UID,
+	exploded []chat1.MessageUnboxed) []chat1.MessageUnboxed {
+	if len(exploded) == 0 {
+		return nil
+	}
+	defer s.Trace(ctx, func() error { return nil }, "getExplodedReplies: num: %d", len(exploded))()
+	var replies []chat1.MessageID
+	for _, msg := range exploded {
+		if !msg.IsValid() {
+			continue
+		}
+		replies = append(replies, msg.Valid().ServerHeader.Replies...)
+	}
+	replyMap := make(map[chat1.MessageID]chat1.MessageUnboxed)
```

Comment:
why not call `updateRepliesAffected` here?

Reply:
Indeed!